pico_w/wifi/access_point/picow_access_point.c

```diff
@@ -10,292 +10,102 @@
+#include "lwip/ip4_addr.h"
+#include "lwip/init.h"
+#include "lwip/apps/httpd.h" // use lwip's httpd server to simplify example
+
+#include "pico/status_led.h" // libary for simplifying onboard LED usage on Pico W
 
 #include "dhcpserver.h"
 #include "dnsserver.h"
 
-#define TCP_PORT 80
 #define DEBUG_printf printf
-#define POLL_TIME_S 5
-#define HTTP_GET "GET"
-#define HTTP_RESPONSE_HEADERS "HTTP/1.1 %d OK\nContent-Length: %d\nContent-Type: text/html; charset=utf-8\nConnection: close\n\n"
-#define LED_TEST_BODY "<html><body><h1>Hello from Pico.</h1><p>Led is %s</p><p><a href=\"?led=%d\">Turn led %s</a></body></html>"
-#define LED_PARAM "led=%d"
-#define LED_TEST "/ledtest"
-#define LED_GPIO 0
-#define HTTP_RESPONSE_REDIRECT "HTTP/1.1 302 Redirect\nLocation: http://%s" LED_TEST "\n\n"
 
-typedef struct TCP_SERVER_T_ {
-    struct tcp_pcb *server_pcb;
-    bool complete;
-    ip_addr_t gw;
-} TCP_SERVER_T;
+static absolute_time_t wifi_connected_time;
+bool led_state;
+bool complete;
 
-typedef struct TCP_CONNECT_STATE_T_ {
-    struct tcp_pcb *pcb;
-    int sent_len;
-    char headers[128];
-    char result[256];
-    int header_len;
-    int result_len;
-    ip_addr_t *gw;
-} TCP_CONNECT_STATE_T;
+static const char *switch_cgi_handler(int iIndex, int iNumParams, char *pcParam[], char *pcValue[]) {
+    printf("Switch cgi handler called\n");
 
-static err_t tcp_close_client_connection(TCP_CONNECT_STATE_T *con_state, struct tcp_pcb *client_pcb, err_t close_err) {
-    if (client_pcb) {
-        assert(con_state && con_state->pcb == client_pcb);
-        tcp_arg(client_pcb, NULL);
-        tcp_poll(client_pcb, NULL, 0);
-        tcp_sent(client_pcb, NULL);
-        tcp_recv(client_pcb, NULL);
-        tcp_err(client_pcb, NULL);
-        err_t err = tcp_close(client_pcb);
-        if (err != ERR_OK) {
-            DEBUG_printf("close failed %d, calling abort\n", err);
-            tcp_abort(client_pcb);
-            close_err = ERR_ABRT;
-        }
-        if (con_state) {
-            free(con_state);
-        }
-    }
-    return close_err;
-}
-
-static void tcp_server_close(TCP_SERVER_T *state) {
-    if (state->server_pcb) {
-        tcp_arg(state->server_pcb, NULL);
-        tcp_close(state->server_pcb);
-        state->server_pcb = NULL;
-    }
-}
+    // Get led state
+    led_state = status_led_get_state();
+    printf("LED state: %d\n", led_state);
 
-static err_t tcp_server_sent(void *arg, struct tcp_pcb *pcb, u16_t len) {
-    TCP_CONNECT_STATE_T *con_state = (TCP_CONNECT_STATE_T*)arg;
-    DEBUG_printf("tcp_server_sent %u\n", len);
-    con_state->sent_len += len;
-    if (con_state->sent_len >= con_state->header_len + con_state->result_len) {
-        DEBUG_printf("all done\n");
-        return tcp_close_client_connection(con_state, pcb, ERR_OK);
+    // Now flip the state
+    if (led_state) {
+        // Turn led off
+        status_led_set_state(0);
+    } else {
+        // Turn led on
+        status_led_set_state(1);
     }
```

Comment:
Or you could just do:
```suggestion
    status_led_set_state(!led_state);
```
?